plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/PluginModernizer.java

```diff
@@ -17,38 +13,20 @@ public class PluginModernizer {
 
     private static final ObjectMapper objectMapper = new ObjectMapper();
 
-    private final MavenGoalInvoker mavenGoalInvoker;
+    private final MavenInvoker mavenInvoker;
 
     public PluginModernizer(Config config) {
         this.config = config;
-        this.mavenGoalInvoker = new MavenGoalInvoker(config);
+        this.mavenInvoker = new MavenInvoker(config);
     }
 
     public void start() {
         String projectRoot = System.getProperty("user.dir");
-        // TODO: Fn to populate the directory with the plugins
-        File pluginsDir = new File(projectRoot, "test-plugins");
-
-        File[] subdirectories = pluginsDir.listFiles(File::isDirectory);
-        if (subdirectories != null) {
-            for (File pluginDir : subdirectories) {
-                try {
-                    List<String> goals = mavenGoalInvoker.createGoalsList();
-                    if (goals == null) {
-                        LOG.info("No active recipes.");
-                        return;
-                    }
-                    mavenGoalInvoker.invokeMavenGoals(pluginDir, goals);
-                } catch (IOException e) {
-                    LOG.error(e.getMessage(), e);
-                }
-            }
-        } else {
-            LOG.error("No subdirectories found in {}", pluginsDir.getAbsolutePath());
+        for (String plugin : config.getPlugins()) {
+            String pluginPath = projectRoot + "/test-plugins/" + plugin;
+            mavenInvoker.invokeGoal(pluginPath, "clean");
```

Comment:
This was just an example, but let's go that way ;) Not sure if a clean phase is needed, but let's got for it